pkg/controller/mcmhub/gitrepo_sync.go

```diff
@@ -71,19 +71,8 @@ func (r *ReconcileSubscription) UpdateGitDeployablesAnnotation(sub *appv1.Subscr
 	if utils.IsGitChannel(string(channel.Spec.Type)) {
 		klog.Infof("Subscription %s has Git type channel.", sub.GetName())
 
-		user, pwd, err := utils.GetChannelSecret(r.Client, channel)
-
-		if err != nil {
-			klog.Errorf("Failed to get secret for channel: %s", channel.GetName())
-			return false, err
-		}
-
-		commit, err := utils.CloneGitRepo(channel.Spec.Pathname,
-			utils.GetSubscriptionBranch(sub),
-			user,
-			pwd,
-			utils.GetLocalGitFolder(channel, sub))
-
+		//making sure the commit id is coming from the same source
+		commit, err := r.hubGitOps.GetLatestCommitID(sub)
```

Comment:
@ianzhang366   In https://github.com/open-cluster-management/multicloud-operators-subscription/pull/313/files#diff-42a6db35407823bbecf54fd8e45c7527R171, the commit ID annotation is updated. That will trigger a reconcile. Then I come here. Wouldn't the comparison of the latest commit ID from hubGitOps to the commit ID annotation be always be the same so the deployables will not be regenerated?

Reply:
good catch, maybe we need to loosen up this one, to make sure the deployable annotation is re-calculated, what do you say

		if !strings.EqualFold(annotations[appv1.AnnotationGitCommit], commit) || r.isHookUpdate(annotations, subKey) {